frontend/packages/dev-console/src/components/edit-application/EditApplicationForm.tsx

```diff
@@ -1,50 +1,63 @@
 import * as React from 'react';
 import * as _ from 'lodash';
 import { FormikProps, FormikValues } from 'formik';
-import { ModalTitle, ModalBody, ModalSubmitFooter } from '@console/internal/components/factory';
-import { BuildStrategyType } from '@console/internal/components/build';
+import { Form, ActionGroup, Button, ButtonVariant } from '@patternfly/react-core';
+import { ButtonBar, PageHeading } from '@console/internal/components/utils';
 import GitSection from '../import/git/GitSection';
 import BuilderSection from '../import/builder/BuilderSection';
 import DockerSection from '../import/git/DockerSection';
 import AdvancedSection from '../import/advanced/AdvancedSection';
 import AppSection from '../import/app/AppSection';
 import { NormalizedBuilderImages } from '../../utils/imagestream-utils';
+import ImageSearchSection from '../import/image-search/ImageSearchSection';
+import { CreateApplicationFlow } from './edit-application-utils';
 
 export interface EditApplicationFormProps {
+  pageHeading: string;
   builderImages?: NormalizedBuilderImages;
 }
 
 const EditApplicationForm: React.FC<FormikProps<FormikValues> & EditApplicationFormProps> = ({
   handleSubmit,
   handleReset,
   values,
+  pageHeading,
   builderImages,
   dirty,
   errors,
   status,
   isSubmitting,
 }) => (
-  <form className="modal-content" onSubmit={handleSubmit}>
-    <ModalTitle>Edit Application</ModalTitle>
-    <ModalBody>
-      {!_.isEmpty(values.build.strategy) && <GitSection />}
-      {values.build.strategy === BuildStrategyType.Source && (
+  <>
+    <PageHeading title={pageHeading} style={{ padding: '0px' }} />
+    <Form onSubmit={handleSubmit}>
+      {pageHeading !== CreateApplicationFlow.Container && <GitSection />}
```

Comment:
All this logic based on `pageHeading` doesn't seem correct semantically. Better name it `createFlow` or `createFlowType` and use it for `title` of the page as well.